enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/CatchUpChannelPool.java

```diff
@@ -92,6 +83,11 @@ private synchronized CHANNEL getIdleChannel( AdvertisedSocketAddress catchUpAddr
         if ( channels != null )
         {
             channel = channels.poll();
+            if ( channel != null && !channel.isActive() )
+            {
+                dispose( channel );
```

Comment:
What would you think about a while-loop here, when the channel !isActive()? e.g. so that we exhaust all suitable idle channels before we return null and cause the creation of a new one

Reply:
Yes! Makes a lot more sense